classes/helper.php

```diff
@@ -155,7 +155,7 @@ protected static function validate_rule_data(\stdClass $formdata): void {
             }
         }
 
-        if (!key_exists($formdata->cohortid, self::get_available_cohorts())) {
+        if (!array_key_exists($formdata->cohortid, self::get_available_cohorts())) {
```

Comment:
I still believe `!isset` here would be better. What are the reasons for preferring this?

```suggestion
        if (!isset(self::get_available_cohorts()[$formdata->cohortid])) {
```

Or even this, since it makes it easier to digest.
```suggestion
        $cohorts = self::get_available_cohorts();
        if (!isset($cohorts[$formdata->cohortid])) {
```

Reply:
Sorry, but I don't think we should change array_key_exists to isset as part of this review. I can't see actual benefits for doing that code wise. Faster means nothing here as we won't measure a difference. And even if we do I don't think it will be big on that code that would make me change it. Also from my experience often isset is a cause of bugs as it returns false on a variable set as null.

Reply:
As alternative I can ask you to raise an new issue for changing that code wise. Where we can discuss it and come to a agreement.

Reply:
No that's okay then. I was more so wondering about your thoughts because you had marked 

Yes I can see it being a relevant of a bug, but that's more often the case because it was misused, rather than because the underlying data changed from something that could not be null to something that could be.

Happy to pass on this, and was more interested in your thoughts on the matter.